app/Filament/Commerce/Resources/CustomerCreditNotes/Pages/EditCustomerCreditNote.php

```
@@ -13,7 +13,9 @@ class EditCustomerCreditNote extends EditRecord
     protected function getHeaderActions(): array
     {
         return [
-            DeleteAction::make(),
+            DeleteAction::make()
+                ->requiresConfirmation()
+                ->visible(fn () => $this->record->status === \App\Enums\Commerce\InvoiceStatus::DRAFT),
```

PHPStan (2.2.8)

[error] Cannot access property $status on Illuminate\Database\Eloquent\Model|int|string.


(property.nonObject)